saleor/webhook/transport/shipping.py

```diff
@@ -64,13 +64,19 @@ def parse_list_shipping_methods_response(
         method_amount = shipping_method_data.get("amount")
         method_currency = shipping_method_data.get("currency")
         method_maximum_delivery_days = shipping_method_data.get("maximum_delivery_days")
+        method_minimum_delivery_days = shipping_method_data.get("minimum_delivery_days")
+        method_description = shipping_method_data.get("description")
+        method_metadata = shipping_method_data.get("metadata")
 
         shipping_methods.append(
             ShippingMethodData(
                 id=to_shipping_app_id(app, method_id),
                 name=method_name,
                 price=Money(method_amount, method_currency),
                 maximum_delivery_days=method_maximum_delivery_days,
+                minimum_delivery_days=method_minimum_delivery_days,
+                description=method_description,
+                metadata=method_metadata,
```

Comment:
There should be some validation of metadata to check if what the app returned follows the format of metadata that Saleor uses. An app could return anything here e.g a null, string, or list. We need to make sure that we handle such cases. Also, I would add tests for resolvers that try to render the GraphQL response with external shipping methods, where app returns both valid and invalid structures as metadata (and other fields).

Reply:
I assume that when invalid structure is returned by the app it should be skipped by the parser and returned as `None` right?

Reply:
I think so, yes. Maybe we could also log a warning in such case, with name of the field that we failed to parsed and the app name.